pfcpiface/p4rt_translator_test.go

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Open Networking Foundation
+
+package pfcpiface
+
+import (
+	"io/ioutil"
+	"testing"
+
+	//nolint:staticcheck // Ignore SA1019.
+	// Upgrading to google.golang.org/protobuf/proto is not a drop-in replacement,
+	// as also P4Runtime stubs are based on the deprecated proto.
+	"github.com/golang/protobuf/proto"
+	p4ConfigV1 "github.com/p4lang/p4runtime/go/p4/config/v1"
+)
+
+//nolint:unused
+func setupNewTranslator() *P4rtTranslator {
+	p4infoBytes, _ := ioutil.ReadFile(p4InfoPath)
+
+	var p4Config p4ConfigV1.P4Info
+
+	_ = proto.UnmarshalText(string(p4infoBytes), &p4Config)
+
+	return newP4RtTranslator(p4Config)
+}
+
+func Test_actionID(t *testing.T) {
+	tests := []struct {
+		name       string
+		args       string
+		translator *P4rtTranslator
+		want       uint32
+	}{
+		{name: "get NoAction",
+			args:       "NoAction",
+			translator: setupNewTranslator(),
+			want:       uint32(21257015),
```

Comment:
We may not have a guarantee for P4Info IDs to remain the same for each new P4Info generated by the p4c compiler. We should figure out a better way to set up expectations.

Reply:
Let's take `actionID` method for instance. I see 2 options:

1. I propose to use an immutable mock p4info file, different than the one in `conf/` to test basic methods like `actionID`, `tableID`. This way we can safely hardcode the IDs.

2. Otherwise use `conf/p4/bin/p4info.txt` but declare another function, within the test, that does basically the same thing and compare it against `actionID`:
```golang
findActionIDFunc := func(p4Info p4ConfigV1.P4Info, name string) uint32 {
		for _, table := range p4Info.Actions {
			if table.GetPreamble().GetName() == name {
				return table.Preamble.Id
			}
		}

		return uint32(0)
	}
```

What do you think?  I'd go for 1

Reply:
As discussed offline, we'll go for auto generated p4 constants